app/models/google_credential.rb

```diff
@@ -13,12 +13,14 @@ def self.from_omniauth(auth_info)
       if credential
         credential
       else
-        registered_user = RegisteredUser.create(status: 1)
-        credential = registered_user.google_credentials.create(google_uid: auth_info.info.google_uid,
-                                                  name: auth_info.info.name,
-                                                  image_url: "",
-                                                  username: auth_info.info.name.parameterize)
-        credential
+        @registered_user = RegisteredUser.create(status: 1)
```

Comment:
I think this should call `#new` instead of `#create` to ensure we don't save any `RegisteredUsers` that don't have proper credentials.